pkg/controller/logforwarding/logforwarding_controller.go

```diff
@@ -0,0 +1,102 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+var log = logf.Log.WithName("controller_logforwarding")
+
+// Add creates a new LogForwarding Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager) error {
+	return add(mgr, newReconciler(mgr))
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	return &ReconcileLogForwarding{client: mgr.GetClient(), scheme: mgr.GetScheme()}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("logforwarding-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource LogForwarding
+	err = c.Watch(&source.Kind{Type: &loggingv1alpha1.LogForwarding{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileLogForwarding implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileLogForwarding{}
+
+// ReconcileLogForwarding reconciles a LogForwarding object
+type ReconcileLogForwarding struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a LogForwarding object and makes changes based on the state read
+// and what is in the LogForwarding.Spec
+//
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileLogForwarding) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling LogForwarding")
+
+	// Fetch the LogForwarding instance
+	instance := &loggingv1alpha1.LogForwarding{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	clf := &loggingv1.ClusterLogForwarder{}
+	if err := instance.ConvertTo(clf); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	opts := []client.CreateOption{}
+	err = r.client.Create(context.TODO(), clf, opts...)
```

Comment:
There should be some status of update of both the GA and TP object here to indicate one is no longer usable and the other was created out of the other